cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/offer/PersistentOfferRequirementProvider.java

```diff
@@ -70,16 +72,42 @@ public OfferRequirement getNewOfferRequirement(CassandraContainer container) {
         }
     }
 
-    public OfferRequirement getReplacementOfferRequirement(Protos.TaskInfo taskInfo) {
-        LOGGER.info("Getting replacement requirement for task: {}", taskInfo.getTaskId().getValue());
-        return getExistingOfferRequirement(taskInfo);
+    public OfferRequirement getReplacementOfferRequirement(CassandraContainer container) {
+        LOGGER.info("Getting replacement requirement for task: {}", container.getId());
+        try {
+            return new OfferRequirement(
+                    clearTaskIds(container.getTaskInfos()),
+                    clearExecutorId(container.getExecutorInfo()),
+                    null,
+                    null);
+        } catch (InvalidRequirementException e) {
+            LOGGER.error("Failed to construct OfferRequirement with Exception: ", e);
+            return null;
+        }
     }
 
     public OfferRequirement getUpdateOfferRequirement(Protos.TaskInfo taskInfo) {
         LOGGER.info("Getting updated requirement for task: {}", taskInfo.getTaskId().getValue());
         return getExistingOfferRequirement(taskInfo);
     }
 
+    private List<Protos.TaskInfo> clearTaskIds(Collection<Protos.TaskInfo> taskInfos) {
+        List<Protos.TaskInfo> outTaskInfos = new ArrayList<>();
+        for (Protos.TaskInfo restartTaskInfo : taskInfos) {
+            outTaskInfos.add(
+                    Protos.TaskInfo.newBuilder(restartTaskInfo)
+                            .setTaskId(Protos.TaskID.newBuilder().setValue(""))
```

Comment:
Ah, clearXxx and hasXxx wont work for required fields, when left unset. Dang!

So this approach seems fine to me, because of our requirements for OfferEvaluator